examples/thermostat/thermostat-common/src/thermostat-delegate-impl.cpp

```diff
@@ -329,7 +339,7 @@ uint8_t ThermostatDelegate::GetUniqueID()
     uint8_t currentId = mUniqueID;
     if (mUniqueID == UINT8_MAX)
     {
-        mUniqueID = 1;
+        mUniqueID = 0;
```

Comment:
OK, but now this is just doing `mUniqueID++`.

Bit you have a bigger problem here, now that I think this through: what guarantees the new value is not already in the list?  What you have to do is look for a value that's not in the list yet.